multiapps-mta/src/main/java/org/cloudfoundry/multiapps/mta/resolvers/v2/ResourceLiveParameterResolver.java

```diff
@@ -0,0 +1,149 @@
+                                                                   .stream()
+                                                                   .anyMatch(defaultParamPlaceholder -> castToString(object).contains(defaultParamPlaceholder));
+    }
+
+    private void replaceValueInResourceParameter(Module module, String resourceParameterValue, Map.Entry<String, Object> resourceParameter,
+                                                 Reference reference, String requiredDependencyValueString) {
+        Map.Entry<String, String> requiredIdleToLiveParameterPair = getMapEntryOnCondition(requiredDependencyValueString);
+        requiredDependencyValueString = requiredDependencyValueString.replace(requiredIdleToLiveParameterPair.getKey(),
+                                                                              castToString(module.getParameters()
+                                                                                                 .get(requiredIdleToLiveParameterPair.getValue())));
+        requiredDependencyValueString = resourceParameterValue.replace(reference.getMatchedPattern(), requiredDependencyValueString);
+
+        resourceParameter.setValue(requiredDependencyValueString);
+    }
+
+    private void resolveModuleProvidedDependency(Map<String, Object> requiredDependencyMap, Module module) {
+        requiredDependencyMap.entrySet()
+                             .forEach(entry -> updateModuleDependencyFromMap(module, entry));
+    }
+
+    private void updateModuleDependencyFromMap(Module module, Map.Entry<String, Object> entry) {
+        if (isStringValueReplaceable(entry.getValue())) {
+            var idleToLiveParameterPair = getMapEntryOnCondition(castToString(entry.getValue()));
+            entry.setValue(castToString(entry.getValue()).replace(idleToLiveParameterPair.getKey(), castToString(module.getParameters()
+                                                                                                                       .get(idleToLiveParameterPair.getValue()))));
+        } else if (entry.getValue() instanceof Map) {
+            resolveModuleProvidedDependency(castToMap(entry.getValue()), module);
+        }
+    }
+
+    private Map.Entry<String, String> getMapEntryOnCondition(String replaceableString) {
+        return idleToLiveParameterPairs.entrySet()
+                                       .stream()
+                                       .filter(idleToLiveParameterPair -> replaceableString.contains(idleToLiveParameterPair.getKey()))
+                                       .findFirst()
+                                       .get();
+    }
+
+    private boolean isModuleNeeded(Module module, String name) {
+        return module.getProvidedDependencies()
+                     .stream()
+                     .anyMatch(dependency -> dependency.getName()
+                                                       .equals(name));
+    }
+
+    private String castToString(Object object) {
```

Comment:
I think this might be less clear than simply using (String) at the appropriate places